include/ginkgo/core/base/executor.hpp

```diff
@@ -536,6 +536,40 @@ class Executor : public log::EnableLogging<Executor> {
             reinterpret_cast<uintptr>(dest_ptr), num_elems * sizeof(T));
     }
 
+    /**
+     * Copies data within this Executor.
+     *
+     * @tparam T  datatype to copy
+     *
+     * @param num_elems  number of elements of type T to copy
+     * @param src_ptr  pointer to a block of memory containing the data to be
+     *                 copied
+     * @param dest_ptr  pointer to an allocated block of memory
+     *                  where the data will be copied to
+     */
+    template <typename T>
+    void copy(size_type num_elems, const T *src_ptr, T *dest_ptr) const
+    {
+        this->copy_from(this, num_elems, src_ptr, dest_ptr);
+    }
+
+    /**
+     * Retrieves a single element at the given location from executor memory.
+     *
+     * @tparam T  datatype to copy
+     *
+     * @param ptr  the pointer to the element to be copied
+     *
+     * @return the value stored at ptr
+     */
+    template <typename T>
+    T copy_to_master(const T *ptr) const
```

Comment:
I would prefer a name that tells that the copy only happens for a single element (so it is not mixed up with `copy`, but for different accessors):
```suggestion
    T copy_element_to_master(const T *ptr) const
```

Or
```suggestion
    T copy_elem_to_master(const T *ptr) const
```

Reply:
Good point! I went with `copy_val_to_host`